### PR TITLE
TLSv1.2

### PR DESCRIPTION
**Pull request by mperazam05**
# Overview

It makes some changes to the session encryption in order to support TLSv1.2.

### CODE REVIEW

File: src/android/Mail.java [hunk 1]
```diff
@@ -196,17 +204,36 @@ public void testConnection() throws MessagingException, NoSuchProviderException
...
+            // upgrade from an insecure connection to a secure one using TLS or SSL.
+            props.put("mail.smtp.starttls.enable", "true");
+            props.put("mail.smtp.socketFactory.port", this.port);
+            if (this.encryption == 1) {
+                props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory"); // SSL factory class
+                props.put("mail.smtp.ssl.protocols", "SSLv3");
+            } else if (this.encryption == 2) {
+                TrustManager[] trustManager = new TrustManager[]{new X509TrustManager() {
+                    public X509Certificate[] getAcceptedIssuers() {
+                        return null;
```
Comment (juanmartin) [inline src/android/Mail.java:221]: Returning `null` means that any issuer is accepted?
Would this mean that any certificate is valid? even locally created ones?
I'm no expert here but if so, seems like an easy thing to fake.

**Comment by juanmartin** [comment]
Kept testing, I used [mailtrap.io](mailtrap.io) as dummy SMTP server.
Looking good 🚀 
image

**Review by juanmartin — APPROVED** [review]

**Comment by bjdose** [comment]
Current config:

From: brayan@raven.com
SMTP Server address: smtp.mailtrap.io
SMTP Port: 2525
Username: Skipped
Password: Skipped
SMTP over SSL/TLS:

I'm not sure why but I got this error when using SSLv3, am I doing something wrong?

image

image


If I use TLSv1.2 it works without any problem.

How could I test the SSLv3?

**Comment by mperazam05** [comment]
This was what I found on the Mailtrap website:

> Yes, you can enforce encrypted connections. Mailtrap SMTP server uses STARTTLS which works for all SMTP ports. We support only TLS connection because of the POODLE vulnerability (SSLv2 and SSLv3 are disabled).

So unfortunately you can't test SSLv3 with that service. I'm still looking for one that supports it because Gmail doesn't seem to either. Let me know if you find any. @bjdose

**Comment by bjdose** [comment]
> This was what I found on the Mailtrap website:
> 
> > Yes, you can enforce encrypted connections. Mailtrap SMTP server uses STARTTLS which works for all SMTP ports. We support only TLS connection because of the POODLE vulnerability (SSLv2 and SSLv3 are disabled).
> 
> So unfortunately you can't test SSLv3 with that service. I'm still looking for one that supports it because Gmail doesn't seem to either. Let me know if you find any. @bjdose

Hi @marvinraven, I was checking the new changes but now I'm getting the same error if I use "TLSv1.2" or "SSLv2, SSLv3, TLSv1.0, TLSv1.1". The only one that works is "None".

The configuration is the same as I commented above. 

Here are the screenshots with the errors.

image

image

image


Google and sendgrid smtp service are working but still having problems with mailtrap.

**Comment by mperazam05** [comment]
> Hi @marvinraven, I was checking the new changes but now I'm getting the same error if I use "TLSv1.2" or "SSLv2, SSLv3, TLSv1.0, TLSv1.1". The only one that works is "None".

I just did a test with Mailtrap and TLSv1.2 is working fine. The others are no longer supported and that's why it's failing you. Let me know if you need us to test together. @bjdose 

Screen Shot 2022-05-02 at 10 22 46 AM

**Review by bjdose — APPROVED** [review]

**Pull request merged**